### PR TITLE
chore: support for building with Java 11 (SDKCF-4919)

### PR DESCRIPTION
**Pull request by leila-gu**
# Description
  - Bumped Java version to Java 11.
  - Bumped gradle to 7.2.
  - Bumped Detekt plugin to 1.20.0.
  - Fixed Dependencycheck vulnerabilities.

## Links
Add links to github/jira issues, design documents and other relevant resources (e.g. previous discussions, platform/tool documentation etc.)

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data **before every commit**, including API endpoints and keys
- [x] I checked that all possible internal and platform/library exceptions are gracefully handled and **will not** crash the host app
- [x] I checked for open & known issues (especially crashes) for any newly added platform APIs or libraries
- [x] I tested non-trivial changes on a real device
- [x] I added ticket/changes in changelog
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: inappmessaging/USERGUIDE.md [hunk 1]
```diff
@@ -387,6 +387,12 @@ Documents targeting Product Managers:
 
 ## <a name="changelog"></a> Changelog
 
+### NEXT (in progress)
```
Comment (vincentdelmundo) [inline inappmessaging/USERGUIDE.md:390]: This should not be breaking change right?

File: inappmessaging/build.gradle [hunk 2]
```diff
@@ -21,15 +21,8 @@ android {
...
-  compileOptions {
-    // Source code java level(Source code of this project).
-    sourceCompatibility JavaVersion.VERSION_1_8
-    // Target JVM level(JVM on Android devices).
-    targetCompatibility JavaVersion.VERSION_1_8
-  }
-
   kotlinOptions {
-    jvmTarget = JavaVersion.VERSION_1_8.toString()
+    jvmTarget = JavaVersion.VERSION_11.toString()
```
Comment (vincentdelmundo) [inline inappmessaging/build.gradle:25]: nit: Why not use the same format as `analytics` module?
`JavaVersion.toVersion(CONFIG.versions.java).toString()`

**Review by vincentdelmundo — CHANGES_REQUESTED** [review]
Looks good. Just a few comments.
* should we also include here the test app changes to support QA test cases?

File: inappmessaging/USERGUIDE.md (hunk 1 above)
Comment (leila-gu) [reply inappmessaging/USERGUIDE.md:390]: Not breaking change,  the host app can continue using the actual config.

File: inappmessaging/build.gradle (hunk 2 above)
Comment (leila-gu) [reply inappmessaging/build.gradle]: updated

**Comment by leila-gu** [comment]
> 

I updated the simple app to persist user info for login-related scenarios. 
No need to cover other scenarios since we can test now using Demoapp!

**Review by vincentdelmundo — CHANGES_REQUESTED** [review]
Looks good. Update the config to latest commit.

**Review by maureenorea-clores — APPROVED** [review]

**Comment by leila-gu** [comment]
> Looks good. Update the config to latest commit.

Updated.

**Review by vincentdelmundo — APPROVED** [review]

**Pull request merged**